javascript/net/grpc/web/grpc_generator.cc

```diff
@@ -120,6 +125,34 @@ void PrintMessagesDeps(Printer* printer, const FileDescriptor* file) {
   printer->Print("\n\n\n");
 }
 
+void PrintCommonJsMessagesDeps(Printer* printer, const FileDescriptor* file) {
+  std::map<string, const Descriptor*> messages = GetAllMessages(file);
+  std::map<string, string> vars;
+  string package = file->package();
+  string filename = file->name();
+  // Remove .proto extension
+  filename = filename.substr(0, filename.size() - 6);
+  vars["package_name"] = package;
+  vars["filename"] = filename;
+
+  printer->Print(vars, "const proto = {};\n");
+  if (!package.empty()) {
+
+    size_t offset = 0;
+    size_t dotIndex = package.find('.');
+
+    while (dotIndex != string::npos) {
+      vars["current_package_ns"] = package.substr(0, dotIndex);
+      printer->Print(vars, "proto.$current_package_ns$ = {};\n");
+
+      offset = dotIndex + 1;
+      dotIndex = package.find(".", offset);
+    }
+  }
+
+  printer->Print(vars, "proto.$package_name$ = require('./$filename$_pb.js');\n\n");
```

Comment:
Does this mean that the proto messages classes have to be generated into one single file named <some name>_pb.js, and can't be generated into one file per message?

Reply:
Yeah I'm matching how the protoc compiler outputs it's message with import_style set to commonjs. They output a js file per proto file with the name `<proto-file-name>_pb.js`.